1. Basics of committing/README.md

```diff
@@ -1 +1,39 @@
-# Basics of committing: understanding the git index
\ No newline at end of file
+# Basics of committing: understanding the git index
+## Overview
+### The git index
+In this section, we will be talking about a few key concepts that relate to how git handles files. The three important places where git handles files: working directory, staging area, and the repository.
+
+When you add code/make changes in your local machine, you are dealing with your project’s working directory; this is synonymous with your machine's filesystem. All the changes you make are considered only existing in the working directory until you add them to the staging area (via `git add` command). The staging area is best described as a preview of your next commit. Meaning, when you run a git commit, git will take the changes that are in the staging area and make the new commit out of those changes. One benefit of the staging area is that it allows you to fine-tune your commits. You can add and remove changes from staging area to your heart's content, after which, you can commit. And after you commit your changes they go into .git/objects directory where they are saved as commit, blob and tree objects. At a high level: blob objects are place-in-time snapshots of files, tree objects represent directories (sort of) as they are composed of blobs and/or other tree objects, and commit objects are pointers to specific tree objects.
+
+Although it is often useful to think of staging area as a tangible directory where git stores changes (like it does in .git/objects) this is not entirely true. Git doesn’t have a dedicated staging directory where it puts some objects representing file changes. Instead, git has a file called the index that it uses to keep track of the file changes over the three areas: working directory, staging area, and repository. And when you add changes to your staging area, git updates the information in the index about those changes and creates new blob objects. It's important to note that the index is a file not a directory; the index keeps track of blobs, trees, and commits via some metadata (basically keeping track of what's in the working directory, staging, and the repo) but does not organized them itself.
```

Comment:
i think this part may be a bit too into weeds for this audience.

Reply:
Yeah wasn't sure how much of the index you wanted to talk about. I'll simplify this down to the working dir, staging, and commits.